Modules/Blueprints/src/selxBlueprintImpl.h

```diff
@@ -30,19 +30,12 @@
 
 // for FromFile and MergeFromFile
 #include <boost/property_tree/ptree.hpp>
-#include <boost/property_tree/xml_parser.hpp>
-#include <boost/property_tree/json_parser.hpp>
-#include <boost/property_tree/info_parser.hpp>
-#include <boost/foreach.hpp>
 #include <boost/filesystem.hpp>
 
 #include <string>
-#include <iostream>
-#include <boost/algorithm/string.hpp>
-
 
 #include "selxBlueprint.h"
-#include "selxLoggerImpl.h"
+#include "selxLogger.h"
```

Comment:
The logger itself is an ITK data object. So with these changes we have a shared pointer to an ITK data object. Not sure if that is compatible with smart pointers? We used the Impl here because both implementations of logger and blueprint are private to the SuperElastix, so the header files are available here, and we could peel of the layer of encapsulation.

Reply:
Sounds indeed dangerous to use shared_ptr and ITK objects together. But does it need to be an ITK object?